File: app/src/main/java/com/sakuna63/sandbox/MainActivity.java
package com.sakuna63.sandbox;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.sakuna63.sandbox.databinding.ActivityMainBinding;

import org.parceler.Parcel;
import org.parceler.ParcelConverter;

import java.util.List;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
    }

    @Parcel(converter = DummyConverter.class)
    interface Interface {

    }

    @Parcel
    static class Data {
        public List<? extends Interface> items;
    }

    static class DummyConverter implements ParcelConverter<Interface> {

        @Override
        public void toParcel(Interface input, android.os.Parcel parcel) {

        }

        @Override
        public Interface fromParcel(android.os.Parcel parcel) {
            return null;
        }
    }

}

File: app/src/main/java/com/sakuna63/sandbox/MainActivity.kt
package com.sakuna63.sandbox

import android.databinding.DataBindingUtil
import android.graphics.Color
import android.os.Bundle
import android.os.Looper
import android.support.v7.app.AppCompatActivity
import android.util.Log
import com.sakuna63.sandbox.databinding.ActivityMainBinding
import kotlin.concurrent.thread

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main)

        binding.root.setOnClickListener {
            Log.d("TAG", "click")
            thread {
                Log.d("TAG", "in new thread")
                if (Looper.myLooper() == null) {
                    Looper.prepare()
                }
                binding.root.setBackgroundColor(Color.BLUE)
//                AlertDialog.Builder(this@MainActivity)
//                        .setTitle("test")
//                        .setMessage("test test test")
//                        .show()
                Log.d("TAG", "finish")
            }
        }
    }
}
